### PR TITLE
BOM-2613: Codemod to remove python_2_unicode_compatible

### PR DESCRIPTION
**Pull request by UsamaSadiq**
**Issue:** [BOM-2613](https://openedx.atlassian.net/browse/BOM-2613)

### Description
- Added code-mod to remove `python_2_unicode_compatible` imports and headers.

## How To Use
- Checkout the branch on local and run the following command: 
- install the package using the command `pip install .`
- refactor any package with the command `remove_python2_unicode_compatible --path {target_files_path}`

## Results
1. 
`from django.utils.encoding import python_2_unicode_compatible`
`from django.utils.translation import ugettext_lazy as _`
-->
` `
`from django.utils.translation import ugettext_lazy as _`

2.
`@python_2_unicode_compatible`
`@login_required`
`class Test:`
--> 
`@login_required`
`class Test:`

### CODE REVIEW

**Review by Jawayria — APPROVED** [review]

File: tests/test_remove_python2_unicode_compatible.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        current_directory = os.path.dirname(__file__)
+        temp_file = os.path.join(current_directory, 'sample_files', str(uuid.uuid4()) + ".py")
+        local_file = os.path.join(current_directory,  'sample_files', file_name)
+        shutil.copy2(local_file, temp_file)
+        return temp_file
+
+    def test_python2_unicode_compatible_removed(self):
+        """
+        Test replace script on a file to make sure it is renaming it properly and
+        also adding request parameter to updated function call
```
Comment (mraarif) [inline tests/test_remove_python2_unicode_compatible.py:28]: we're not adding `request` parameter in updated function parameters, are we?

Comment (UsamaSadiq) [reply tests/test_remove_python2_unicode_compatible.py]: Updated the description.

File: tests/test_remove_python2_unicode_compatible.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import uuid
+import subprocess
+
+
+class Python2UnicodeCompatibleTests(TestCase):
+    """
+    Tests for the Python2_unicode_compatible.py output
+    """
+    def setup_local_copy(self, file_name):
+        """
```
Comment (mraarif) [inline tests/test_remove_python2_unicode_compatible.py:16]: we should instead use `pytest`'s `tmpdir` , as used here https://github.com/edx/repo-tools/blob/1594e55dd4439d8860d203c478cd62008df0de8c/tests/test_modernize_openedx_yaml.py#L9

File: tests/test_remove_python2_unicode_compatible.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    def test_python2_unicode_compatible_removed(self):
+        """
+        Test the script on a sample file to make sure it is removing python_2_unicode_compatible imports and decorators.
+        """
+        sample_file_path = self.setup_local_copy("sample_python2_unicode_removal.py")
+        with open(sample_file_path) as sample_file:
+            sample_code = sample_file.read()
+            assert "python_2_unicode_compatible" in sample_code
+        subprocess.call(['remove_python2_unicode_compatible', sample_file_path])
```
Comment (mraarif) [inline tests/test_remove_python2_unicode_compatible.py:33]: why don't we call the modernizer script directly here

File: tests/test_remove_python2_unicode_compatible.py (hunk 2 above)
Comment (UsamaSadiq) [reply tests/test_remove_python2_unicode_compatible.py]: Updated the function as per suggestion.

File: tests/test_remove_python2_unicode_compatible.py (hunk 3 above)
Comment (UsamaSadiq) [reply tests/test_remove_python2_unicode_compatible.py]: Updated the function implementation according to the suggestion.

File: edx_repo_tools/codemods/django3/remove_python2_unicode_compatible.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        .select("decorator<'@' name='python_2_unicode_compatible' any>")
+        .modify(remove_node)
+        .select("import_from<'from' module_name=any 'import' 'python_2_unicode_compatible'>")
+        .modify(remove_node)
+        .write()
+    )
+
+
+@click.command()
+@click.option('--path', help='use syntax: --path {path_to_test_file}')
```
Comment (mraarif) [inline edx_repo_tools/codemods/django3/remove_python2_unicode_compatible.py:31]: shouldn't this also accept directory paths? and iterate over python recursively and remove `python_2_unicode_compatible` decorator

Comment (UsamaSadiq) [reply edx_repo_tools/codemods/django3/remove_python2_unicode_compatible.py:31]: It works with the directory path too and refactors all files recursively. I'll update the description.

**Review by mraarif — APPROVED** [review]

**Pull request merged**